examples/esp8266/esphub/main.go

```diff
@@ -0,0 +1,103 @@
+// This is a sensor hub that uses a ESP8266 running on the device UART1.
+// It creates a UDP "server" you can use to get info to/from your computer via the microcontroller.
+//
+// In other words:
+// Your computer <--> UART0 <--> MCU <--> UART1 <--> ESP8266 <--> INTERNET
+//
+package main
+
+import (
+	"machine"
+	"time"
+
+	"github.com/tinygo-org/drivers/esp8266"
+)
+
+// change actAsAP to true to act as an access point instead of connecting to one.
+const actAsAP = false
+
+// access point info
+const ssid = "YOURSSID"
+const pass = "YOURPASS"
+
+// change these to connect to a different UART or pins for the ESP8266
+var (
+	uart       = machine.UART1
+	tx   uint8 = machine.D10
+	rx   uint8 = machine.D11
+
+	console = machine.UART0
+
+	adaptor *esp8266.Device
+)
+
+func main() {
+	uart.Configure(machine.UARTConfig{TX: tx, RX: rx})
+
+	// Init esp8266
+	dev := esp8266.New(uart)
+	adaptor = &dev
+	adaptor.Configure()
+
+	// first check if connected
+	if adaptor.Connected() {
+		console.Write([]byte("Connected to ESP8266.\r\n"))
+		adaptor.Echo(false)
+
+		if actAsAP {
+			provideAP()
+		} else {
+			connectToAP()
+		}
+	} else {
+		console.Write([]byte("\r\n"))
+		console.Write([]byte("Unable to connect to esp8266.\r\n"))
+		return
+	}
+
+	// now make UDP connection
+	laddr := &esp8266.UDPAddr{Port: 2222}
+	console.Write([]byte("Loading UDP listener...\r\n"))
+	conn, _ := adaptor.ListenUDP("UDP", laddr)
+
+	console.Write([]byte("Waiting for data...\r\n"))
+	data := make([]byte, 50)
+	for {
+		n, _ := conn.Read(data)
+		if n > 0 {
+			println(string(data))
```

Comment:
This prints all data, even the null bytes (which your console probably hides). You can print `string(data[:n])` to avoid that, and to avoid having to zero the slice afterwards.
Also, why `println` here when you use `console.Write` elsewhere?